src/libraries/Microsoft.PowerFx.Core/Parser/ParseFormulasResult.cs

```diff
@@ -78,16 +83,23 @@ internal class UDF
 
         internal bool IsImperative { get; }
 
-        internal bool NumberIsFloat { get; }
+        internal bool NumberIsFloat { get; }
+
+        /// <summary>
+        /// False if UDF is incomplete eg: Add(a: Number, b: Number): .
+        /// </summary>
+        internal bool IsParseValid { get; }
 
-        public UDF(IdentToken ident, IdentToken returnType, HashSet<UDFArg> args, TexlNode body, bool isImperative, bool numberIsFloat)
+        public UDF(IdentToken ident, Token colonToken, IdentToken returnType, HashSet<UDFArg> args, TexlNode body, bool isImperative, bool numberIsFloat, bool isValid)
         {
             Ident = ident;
             ReturnType = returnType;
             Args = args;
             Body = body;
             IsImperative = isImperative;
-            NumberIsFloat = numberIsFloat;
+            NumberIsFloat = numberIsFloat;
+            IsParseValid = isValid;
+            ReturnTypeColonToken = colonToken;
```

Comment:
>ReturnTypeColonToken = colonToken; [](http://example.com/codeflow?start=12&length=34)

Do you have a draft PR on how this would be used in paclient?

Reply:
Yes